File: benchmarks/node/string_scan.js
const text = Buffer.from("TezzNative makes native tools readable and fast.");
let acc = 0;
for (let i = 0; i < 60000; i += 1) {
  for (let j = 0; j < text.length; j += 1) {
    acc = (acc + ((text[j] * (j + 1)) & 65535)) & 1048575;
  }
}
if (acc === -1) {
  process.exit(1);
}

File: benchmarks/rust/string_scan.rs
fn main() {
    let text = b"TezzNative makes native tools readable and fast.";
    let mut acc: i32 = 0;
    for _ in 0..60_000 {
        for (j, ch) in text.iter().enumerate() {
            acc = (acc + (((*ch as i32) * ((j as i32) + 1)) & 65_535)) & 1_048_575;
        }
    }
    if acc == -1 {
        std::process::exit(1);
    }
}
